autofs-5.0.8 - allow --with-systemd to take a path arg

From: Joe MacDonald <joe@deserted.net>

If building for a cross-compile environment with systemd it is convenient
to be able to specify a systemd path for the target that may not be the
same as that on the host.

I encountered a problem while working with autofs in a cross-compile
environment where the host and target have either differing systemd paths
or where one may not have systemd support at all.  The common solution
I've seen in other projects is to have a --with-systemddir=[path] option,
but I thought it'd be simpler to add an optional path argument to the
extant '--with-systemd' configure parameter and leave the default =y+probe
for path logic in place.

Signed-off-by: Joe MacDonald <joe@deserted.net>
---
 CHANGELOG  |    1 +
 aclocal.m4 |   10 ++++++++--
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/CHANGELOG b/CHANGELOG
index a01393c..565153d 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -3,6 +3,7 @@
 - fix undefined authtype_requires_creds err if ldap enabled but without sasl.
 - fix master map type check.
 - fix task manager not getting signaled.
+- allow --with-systemd to take a path arg.
 
 17/10/2013 autofs-5.0.8
 =======================
diff --git a/aclocal.m4 b/aclocal.m4
index 3e6f223..105e3e9 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -229,8 +229,10 @@ dnl Check the location of the systemd unit files directory
 dnl --------------------------------------------------------------------------
 AC_DEFUN([AF_WITH_SYSTEMD],
 [AC_ARG_WITH(systemd,
-[  --with-systemd         install systemd unit file if systemd unit directory
-			  is found on system],
+[  --with-systemd@<:@=systemddir@:>@  install systemd unit file.  If 'yes'
+			  probe the system for unit directory.
+			  If a path is specified, assume that
+			  is a valid install path.],
 [if test "$withval" = yes; then
   if test -z "$systemddir"; then
     AC_MSG_CHECKING([location of the systemd unit files directory])
@@ -247,6 +249,10 @@ AC_DEFUN([AF_WITH_SYSTEMD],
   else
     AC_MSG_RESULT(not found)
   fi
+else
+ if test "$withval" != no; then
+  systemddir=$withval
+ fi
 fi])
 ])
 
